routers/routes/gin.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package routes
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"path"
+	"text/template"
+	"time"
+
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/public"
+	"code.gitea.io/gitea/modules/setting"
+
+	"github.com/gin-gonic/gin"
+)
+
+type routerLoggerOptions struct {
+	Ctx            *gin.Context
+	Identity       *string
+	Start          *time.Time
+	ResponseWriter gin.ResponseWriter
+}
+
+// SignedUserName returns signed user's name via context
+// FIXME currently no any data stored on gin.Context but macaron.Context, so this will
+// return "" before we remove macaron totally
+func SignedUserName(ctx *gin.Context) string {
+	if v, ok := ctx.Get("SignedUserName"); !ok {
+		return ""
+	} else {
```

Comment:
@kolaente golangci-lint failed because 
```
routers/routes/gin.go:35:9: if block ends with a return statement, so drop this else and outdent its block (move short variable declaration to its own line if necessary) (golint)
--
583 | } else {
584 | ^
585 | Makefile:471: recipe for target 'golangci-lint' failed
```

But I don't think there is another better style. :(

Reply:
```suggestion
	}
```

Reply:
What about something like
```go
var uname string
if v, ok := ctx.Get("SignedUserName"); ok {
  uname = v.(string)
}
return uname
```

Reply:
What about something like
```go
var uname string
if v, ok := ctx.Get("SignedUserName"); ok {
  uname = v.(string)
}
return uname
```

Reply:
@lunny Exactly what @jolheiser meant is needed here.